### PR TITLE
refactor: spectra and auto assignments panel bar buttons

### PR DESCRIPTION
**Pull request by hamed-musallam**
- [x] refactor the spectra and auto assignments panel bar buttons
- [x] delete/show/hide spectra based on the selected spectra
- [x] select all the spectra if the mouse over the spectra table and press Ctrl + a 
- [x] prevent show spectra when select spectrum/spectra are hidden  
- [x] Fix tests

### CODE REVIEW

**Comment by lpatiny** [comment]
Rather that stealing the focus (that leads to bad side effect like if you move your mouse when editing a peak chemical shift) it would be better to assign globally 

CTRL (CMD on mac) + A to select all spectra. This can even work if the spectra panel is not open.

**Comment by hamed-musallam** [comment]
> Rather that stealing the focus (that leads to bad side effect like if you move your mouse when editing a peak chemical shift) it would be better to assign globally
> 
> CTRL (CMD on mac) + A to select all spectra. This can even work if the spectra panel is not open.

The problem we use the same shortcut for other things, especially for Phase correction, and Apodization

**Comment by lpatiny** [comment]
For phase correction and apodization we don't use any modifiers key.

So CTRL (CMD) A is currently not use and it selects all the text in the browser by default.

**Comment by hamed-musallam** [comment]
the side effect is acceptable because in any way each time you enter over the table, we get the focus and vice versa

So if we take your example, when we edit the  "peak chemical shift" we lost focus, this is normal behavior and I think is acceptable because it does not make sense to have a "select all" option and I'm not on the spectra table

**Comment by hamed-musallam** [comment]
@lpatiny 

the apodization and phase correction triggered because we only check if !e.shiftKey && !e.metaKey

**Comment by hamed-musallam** [comment]
@lpatiny 

As i mentioned, if we use this event globally then the only way to know whether I'm on the spectra list or not is by looking inside the objects list on which listeners will be invoked `composedPath`.

 from my point of view, it does not make sense to trigger the select all and the mouse is not over the spectra table

**Comment by lpatiny** [comment]
On my computer CMD + A select the page so I don't understand your answer.



Currently taking the focus leads to bugs.

**Comment by lpatiny** [comment]
It should work where ever is the mouse so in fact it simplifies the code.

**Comment by hamed-musallam** [comment]
@lpatiny 

it triggers the phase correction for me 
image

**Comment by hamed-musallam** [comment]
@lpatiny 

> Currently taking the focus leads to bugs.

Which kind of problem do we have when wee lost focus?
